R/addAxis.R

```diff
@@ -0,0 +1,55 @@
+#' Add Spatial Cardinal Directions to Axis Labels
+#'
+#' This function adds cardinal directions to axis labels in a map.
+#'
+#' @param side An integer between 1 (bottom x-Axis) and 4 (right y-Axis).
+#' @param at Values on the axis to add labels.
+#' @param mgp Customize axis positions. See `par()`.
+#' @param digits Number of digits of labels.
+#' @param ... Other parameters to pass to `axis()`.
+#'
+#' @importFrom graphics par axTicks axis
+#' @export
+#'
+#' @keywords axis, spatial
+#'
+#' @author Nicolas CASAJUS, \email{nicolas.casajus@@gmail.com}
+#'
+#' @examples
+#' maps::map()
+#' addFrame(grid = TRUE, add = TRUE, width = 5)
+#' addAxis(1)
+#' addAxis(2, at = seq(-50, 50, 100), digits = 2)
+#' addAxis(3, lwd = 0, lwd.ticks = 0.5)
+#' addAxis(4, las = 1, col.axis = "red")
+
+
+addAxis <- function(side, at = axTicks(side), mgp = par()$mgp, digits = 0,
+                     ...) {
+
+  opar <- par(no.readonly = TRUE)
+
+  options(warn = -1)
+  par(mgp = mgp)
+  options(warn = 0)
+
```

Comment:
If you change do that I would make sure that the user preferences are not affected. Let say a user is using `warn=-1` after calling this function his option would have been changed, which is not desired!

Reply:
Yeah, need to deal as `par()`, catch user values ans restore it on exit

Reply:
Done.
Adding `owarn <- options()$warn` and `on.exit(options(warn = owarn))`